cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -299,7 +301,12 @@ private void checkInitialized() {
       Preconditions.checkNotNull(scheduler, "Scheduler not yet initialized");
     }
 
-    private JobKey getJobKey(Id.Program program, ProgramType programType) {
+    private String getScheduleId(Id.Program program, SchedulableProgramType programType, String scheduleName) {
+      return String.format("%s:%s", getJobKey(program, programType).getName(), scheduleName);
+    }
+
+
+    private JobKey getJobKey(Id.Program program, SchedulableProgramType programType) {
       return new JobKey(String.format("%s:%s:%s:%s", programType.name(), program.getNamespaceId(),
```

Comment:
I think we can remove ProgramType from the key. it'll simplify the API
considerably.

On Fri, Jan 16, 2015 at 6:16 PM, Andreas Neumann <notifications@github.com>
wrote:

> In
> cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java
> <https://github.com/caskdata/cdap/pull/907#discussion-diff-23121703>:
>
> > @@ -299,7 +301,12 @@ private void checkInitialized() {
> >        Preconditions.checkNotNull(scheduler, "Scheduler not yet initialized");
> >      }
> >
> > -    private JobKey getJobKey(Id.Program program, ProgramType programType) {
> > +    private String getScheduleId(Id.Program program, SchedulableProgramType programType, String scheduleName) {
> > +      return String.format("%s:%s", getJobKey(program, programType).getName(), scheduleName);
> > +    }
> > +
> > +
> > +    private JobKey getJobKey(Id.Program program, SchedulableProgramType programType) {
> >        return new JobKey(String.format("%s:%s:%s:%s", programType.name(), program.getNamespaceId(),
>
> normally we have namespace/app/type/program. Why use a different order
> here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caskdata/cdap/pull/907/files#r23121703>.
>